src/DataProvider/OperationDataProviderTrait.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataProvider;
+
+use ApiPlatform\Core\Exception\InvalidIdentifierException;
+use ApiPlatform\Core\Identifier\Normalizer\ChainIdentifierDenormalizer;
+
+trait OperationDataProviderTrait
+{
+    /**
+     * @var CollectionDataProviderInterface
+     */
+    private $collectionDataProvider;
```

Comment:
Please add a blank line between props (btw, it's often better to have traits without props).